src/main/java/uk/gov/hmcts/divorce/common/event/Applicant2SolicitorApplyForFinalOrder.java

```diff
@@ -0,0 +1,239 @@
+            new Applicant2SolApplyForFinalOrderDetails(),
+            new Applicant2SolFinalOrderExplainWhyNeedToApply(),
+            solFinalOrderPayment,
+            new HelpWithFeesPageForApplicant2SolFinalOrder(),
+            new SolFinalOrderPayAccount(),
+            new SolFinalOrderPaymentSummary()
+        );
+
+        final PageBuilder pageBuilder = addEventConfig(configBuilder);
+        pages.forEach(page -> page.addTo(pageBuilder));
+    }
+
+    private PageBuilder addEventConfig(ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        return new PageBuilder(configBuilder
+            .event(FINAL_ORDER_REQUESTED_APP2_SOL)
+            .forStates(AwaitingFinalOrder)
+            .name(APPLY_FOR_FINAL_ORDER)
+            .description(APPLY_FOR_FINAL_ORDER)
+            .showCondition("applicationType=\"soleApplication\" AND finalOrderReminderSentApplicant2=\"Yes\"")
+            .showSummary()
+            .showEventNotes()
+            .grant(CREATE_READ_UPDATE, APPLICANT_2_SOLICITOR)
+            .aboutToStartCallback(this::aboutToStart)
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .submittedCallback(this::submitted)
+            .grantHistoryOnly(
+                CASE_WORKER,
+                SUPER_USER,
+                LEGAL_ADVISOR,
+                APPLICANT_1_SOLICITOR,
+                JUDGE));
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(final CaseDetails<CaseData, State> details) {
+        log.info("{} about to start callback invoked for Case Id: {}", FINAL_ORDER_REQUESTED_APP2_SOL, details.getId());
+
+        CaseData data = details.getData();
+        log.info("Retrieving order summary");
+        final OrderSummary orderSummary = paymentService.getOrderSummaryByServiceEvent(SERVICE_OTHER, EVENT_GENERAL, KEYWORD_NOTICE);
+
+        data.getFinalOrder().setApplicant2SolFinalOrderFeeOrderSummary(orderSummary);
+
+        data.getFinalOrder().setApplicant2SolFinalOrderFeeInPounds(
+            NumberFormat.getNumberInstance().format(
+                new BigDecimal(orderSummary.getPaymentTotal()).movePointLeft(2)
+            )
+        );
+
+        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+            .data(data)
+            .errors(null)
+            .warnings(null)
+            .build();
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<CaseData, State> details,
+                                                                       CaseDetails<CaseData, State> beforeDetails) {
+
+        log.info("{} about to submit callback invoked for Case Id: {}", FINAL_ORDER_REQUESTED_APP2_SOL, details.getId());
+
+        CaseData data = details.getData();
+        data.getApplication().setPreviousState(beforeDetails.getState());
+
+        final List<String> errors = applyForFinalOrderService.validateApplyForFinalOrder(data, true);
```

Comment:
Will this catch anything for the applicant2 solicitor event? The validation method checks if the "applicant1AppliedFirst" or "applicant2AppliedFirst" vars are set. Do we set them for this flow?

Reply:
Removed validateApplyForFinalOrder call - only applicable for joint cases